backend/src/test/java/capstone/storage/backend/ItemServiceTest.java

```diff
@@ -0,0 +1,206 @@
+package capstone.storage.backend;
+
+import capstone.storage.backend.exceptions.ExceptionMessage;
+import capstone.storage.backend.exceptions.ItemAlreadyExistException;
+import capstone.storage.backend.item.ItemEanApiService;
+import capstone.storage.backend.item.ItemRepo;
+import capstone.storage.backend.item.ItemService;
+import capstone.storage.backend.item.models.AddItemDto;
+import capstone.storage.backend.item.models.Item;
+import capstone.storage.backend.item.models.ItemResponse;
+import capstone.storage.backend.utils.ServiceUtils;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.*;
+
+class ItemServiceTest {
+    private final ItemRepo itemRepo = mock(ItemRepo.class);
+    private final ServiceUtils utils = mock(ServiceUtils.class);
+    private final ItemEanApiService eanApiService = mock(ItemEanApiService.class);
+    private final ItemService itemService = new ItemService(itemRepo, eanApiService, utils);
+
+    @Test
+    @DisplayName("method : findAll -> should return an empty list")
+    void findAllItemsAndExpectEmptyList() {
+        //GIVEN
+        List<Item> itemList = new ArrayList<>();
+        when(itemRepo.findAll()).thenReturn(itemList);
+        //WHEN
+        List<Item> actual = itemService.findAll();
+        List<Item> expected = itemList;
```

Comment:
Dieser Test schließt nicht aus, dass der itemService in die Liste, die er vom Repo bekommt, Daten einfügt und diese gefüllte Liste returned.
Deshalb besser eine Immutable-Liste nutzen: Dann ist expected sicher auch nach der Ausführung von findAll leer.
```suggestion
        List<Item> itemList = Collections.emptyList();
        when(itemRepo.findAll()).thenReturn(itemList);
        //WHEN
        List<Item> actual = itemService.findAll();
        List<Item> expected = itemList;
```